### ISSUE TITLE
toktx: failed to create image from <img> Format of input file is unsupported

### ISSUE THREAD

**Issue by juber-ivre** [KhronosGroup/KTX-Software#441]
Love the tool, very handy for KTX conversions. The issue I'm facing is this, trying to convert the image below and getting unsupported input file.

Does anyone would know why?

![Top-view-Closeup-blue-water-ringsYJ-1](https://user-images.githubusercontent.com/48788224/122576849-10074000-d04a-11eb-99e9-b2eda6695eaf.jpg)

**Comment by MarkCallow** [comment]
Despite the name it has been given in your comment, it appears to be a .webp image which are not supported by `toktx`. When compressing to Basis universal (or any GPU compressed texture format) it is stringkly recommended to start from uncompressed images. WebP images are compressed.

**Comment by lexaknyazev** [comment]
@MarkCallow 
WebP has both lossy and lossless modes so we may consider adding it as a supported input format if there's enough demand.

**Comment by alexkowel** [comment]
Indeed, having this option supported would be very useful. We develop Verge3D, a WebGL toolkit that among other things have strong integration with Blender. Blender just got support for WebP images and our users are asking whether they can use KTX texture compression for this very promising format.

**Comment by MarkCallow** [comment]
@alexkowel what do your users want to do with webp and what makes it "promising" for it?

Note that even if webp is used to deliver a texture to a device, it will be fully inflated on the GPU and thus save no memory. The advantages of using webp for source images for `toktx`, if any, depend on its usefulness in the rest of the content creation pipeline. Within `toktx` it will be inflated and the result used directly, if an uncompressed output format is selected, or re-encoded with BasisU or ASTC, if one of those output formats is selected. Therefore any improvements over, say, JPEG source images will be marginal.

**Comment by alexkowel** [comment]
@MarkCallow I agree that switching to WebP won't affect compressed images. The rationale behind my suggestion is as follows.

1) Blender artists just want to use WebP because it's supported natively in Blender and have many advantages over PNG/JPG.
2) We don't really want to force users to not use this format if they need KTX compression.
3) If KTX compression were supported for WebP, the artists could just use this efficient format everywhere and select  compression individually for each image at their discretion, e.g disable compression if any artifacts found.

**Comment by MarkCallow** [comment]
> 3\. If KTX compression were supported for WebP,

What exactly do you mean? If you mean KTX files containing WebP compressed images, that isn't going to happen. KTX is for GPU ready formats.

**Comment by MarkCallow** [comment]
> On the other hand, encoding lossless WebP textures to KTX/BasisU is fine and therefore should not require any extra steps.

When support is added how should we handle lossy WebP images? Should we warn or error? My opinion is warn.

**Comment by MarkCallow** [comment]
Changed title as the legacy tool `toktx` will not be updated but we may add this feature to the new ktx tool suite in future.
